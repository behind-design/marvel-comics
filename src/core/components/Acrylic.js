import React from 'react';


const addImage = (styles, image) => ({
  ...styles,
  background: `#000 url(${image}) no-repeat center/100%`,
});

export default ({
  image,
  children,
}) => (
  <div className="marvel-comics-acrylic" style={styles.wrapper}>
    <div className="marvel-comics-acrylic__container" style={addImage(styles.container, image)}>
      <div className="marvel-comics-acrylic__before" style={addImage(styles.before, image)}></div>
      <div className="marvel-comics-acrylic__after" style={styles.after}></div>
    </div>
    {children}
  </div>
)

const styles = {};

styles.wrapper = {
  position: 'relative',
  width: '100%',
  height: '100%',
}

styles.container = {
  width: '100%',
  height: '100%',
  position: 'absolute',
  backgroundSize: 'cover',
  overflow: 'hidden',
}

styles.before = {
  // width: 'calc(100% + 50px)',
  // height: 'calc(100% + 50px)',
  width: 'calc(100% + 75px)',
  height: 'calc(100% + 75px)',
  position: 'absolute',
  top: '-25px',
  // left: '-25px',
  left: '-33px',
  backgroundBlendMode: 'exclusion',
  backgroundSize: 'cover',
  filter: 'blur(25px)',
}

styles.after = {
  height: '100%',
  width: '100%',
  position: 'absolute',
  top: 0,
  left: 0,
  right: 0,
  bottom: 0,
  boxShadow: '0 0 3px rgba(#000 .1)',
  backgroundImage: `url("http://api.thumbr.it/whitenoise-361x370.png?background=ffffff00&noise=a0a0a0&density=35&opacity=10")`,
  backgroundColor: 'rgba(0, 0, 0, .5)',
}
