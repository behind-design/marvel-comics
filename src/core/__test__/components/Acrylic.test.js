import React from 'react';
import ReactDOM from 'react-dom';
import { mount, render } from 'enzyme';
import renderer from 'react-test-renderer';

import Acrylic from '../../components/Acrylic';


const IMAGE_NOT_FOUND = `http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_uncanny.jpg`;

describe('Acrylic', () => {
  it('renders without crashing', () => {
    const json = renderer.create(
      <Acrylic image={IMAGE_NOT_FOUND} />
    ).toJSON();

    expect(json).toMatchSnapshot();
  });

  it('renders all the correct properties', () => {
    const wrapper = renderer.create(
      <Acrylic image={IMAGE_NOT_FOUND} />
    ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
