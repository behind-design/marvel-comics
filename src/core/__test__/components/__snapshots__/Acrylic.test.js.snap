// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Acrylic renders all the correct properties 1`] = `
<div
  className="marvel-comics-acrylic"
  style={
    Object {
      "height": "100%",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    className="marvel-comics-acrylic__container"
    style={
      Object {
        "background": "#000 url(http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_uncanny.jpg) no-repeat center/100%",
        "backgroundSize": "cover",
        "height": "100%",
        "overflow": "hidden",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <div
      className="marvel-comics-acrylic__before"
      style={
        Object {
          "background": "#000 url(http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_uncanny.jpg) no-repeat center/100%",
          "backgroundBlendMode": "exclusion",
          "backgroundSize": "cover",
          "filter": "blur(25px)",
          "height": "calc(100% + 75px)",
          "left": "-33px",
          "position": "absolute",
          "top": "-25px",
          "width": "calc(100% + 75px)",
        }
      }
    />
    <div
      className="marvel-comics-acrylic__after"
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, .5)",
          "backgroundImage": "url(\\"http://api.thumbr.it/whitenoise-361x370.png?background=ffffff00&noise=a0a0a0&density=35&opacity=10\\")",
          "bottom": 0,
          "boxShadow": "0 0 3px rgba(#000 .1)",
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;

exports[`Acrylic renders without crashing 1`] = `
<div
  className="marvel-comics-acrylic"
  style={
    Object {
      "height": "100%",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    className="marvel-comics-acrylic__container"
    style={
      Object {
        "background": "#000 url(http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_uncanny.jpg) no-repeat center/100%",
        "backgroundSize": "cover",
        "height": "100%",
        "overflow": "hidden",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <div
      className="marvel-comics-acrylic__before"
      style={
        Object {
          "background": "#000 url(http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_uncanny.jpg) no-repeat center/100%",
          "backgroundBlendMode": "exclusion",
          "backgroundSize": "cover",
          "filter": "blur(25px)",
          "height": "calc(100% + 75px)",
          "left": "-33px",
          "position": "absolute",
          "top": "-25px",
          "width": "calc(100% + 75px)",
        }
      }
    />
    <div
      className="marvel-comics-acrylic__after"
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, .5)",
          "backgroundImage": "url(\\"http://api.thumbr.it/whitenoise-361x370.png?background=ffffff00&noise=a0a0a0&density=35&opacity=10\\")",
          "bottom": 0,
          "boxShadow": "0 0 3px rgba(#000 .1)",
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
